controllers/utils/predicate.go

```diff
@@ -54,16 +56,21 @@ func CreateRequests(client runtimeclient.Client) handler.EventHandler {
 			if err := client.List(context.TODO(), &vgList); err != nil {
 				return []ctrl.Request{}
 			}
-			// TODO CSI-5437 - add a label selector check to the VolumeGroup to filter the list
 			// Create a reconcile request for each matching VolumeGroup.
-			requests := make([]ctrl.Request, len(vgList.Items))
+			var requests []ctrl.Request
 			for _, vg := range vgList.Items {
-				requests = append(requests, ctrl.Request{
-					NamespacedName: types.NamespacedName{
-						Namespace: vg.Namespace,
-						Name:      vg.Name,
-					},
-				})
+				selector, err := metav1.LabelSelectorAsSelector(vg.Spec.Source.Selector)
+				if err != nil {
+					return []ctrl.Request{}
+				}
+				if selector.Matches(labels.Set(object.GetLabels())) {
```

Comment:
can you try and use [this](https://github.com/IBM/csi-volume-group-operator/blob/develop/controllers/utils/labels.go#L35-L38) function?

Reply:
done ( used areLabelsMatchLabelSelector)